src/python/zquantum/core/estimator.py

```diff
@@ -102,23 +106,42 @@ def get_estimated_expectation_values(
         circuit: Circuit,
         target_operator: SymbolicOperator,
         n_samples: Optional[int] = None,
+        n_total_samples: Optional[int] = None,
         epsilon: Optional[float] = None,
         delta: Optional[float] = None,
+        shot_allocation_strategy: str = "uniform",
     ) -> ExpectationValues:
         """Given a circuit, backend, and target operators, this method produces expectation values
         for each target operator using the get_expectation_values method built into the provided QuantumBackend.
 
         Args:
-            backend (QuantumBackend): the backend that will be used to run the circuit
-            circuit (Circuit): the circuit that prepares the state.
-            target_operator (List[SymbolicOperator]): List of target functions to be estimated.
-            n_samples (int): Number of measurements done.
-            epsilon (float): an error term.
-            delta (float): a confidence term.
+            backend: the backend that will be used to run the circuit
+            circuit: the circuit that prepares the state.
+            target_operator): List of target functions to be estimated.
+            n_samples: Number of measurements to be performed on each frame.
+                Exactly one of n_samples and n_total_samples must be provided.
+            n_total_samples: Total number of measurements to be performed across
+                all frames. Exactly one of n_samples and n_total_samples must be
+                provided.
+            epsilon: Inherited from Estimator, not used.
```

Comment:
I think that if `epsilon` and `delta` are just `Inherited from Estimator, not used` – we should change the `Estimator` interface so that they're are not required.
I had issues with these elsewhere too and they don't think like a good part of the `Estimator` interface.

Reply:
I think that `epsilon` (desired precision) might make sense to keep, although it could use a more descriptive name. We could make `BasicEstimator` support it, just haven't gotten around to it yet. I'm less certain that we should keep `delta`, which I think @sabeegrewal originally intended to use for specifying the desired bound on the error according to some confidence level.

Reply:
Ok, I'll defer to your judgment here, as I don't have a good feeling about what would be best.
Also, it might be reasonable to deal with that in a separate PR.